3-JSS/scripts/localstorage_manager.js

```diff
@@ -0,0 +1,37 @@
+const LOCAL_STORAGE_KEY = "myChecklist";
+
+function setMapToLocalStorage(map) {
+  localStorage.setItem(
+    LOCAL_STORAGE_KEY,
+    JSON.stringify(Array.from(map.entries()))
+  );
+}
+
+function getMapFromLocalStorage() {
+  let jsonObject = JSON.parse(localStorage.getItem(LOCAL_STORAGE_KEY));
+  let map = jsonObject != null ? new Map(jsonObject) : new Map();
```

Comment:
consider using a regular json object so you won't have to do this "complex" serialize/deserialize every time